fn/slice.go

```diff
@@ -205,3 +209,24 @@ func Sum[B Number](items []B) B {
 func HasDuplicates[A comparable](items []A) bool {
 	return len(NewSet(items...)) != len(items)
 }
+
+// ForEachConc maps the argument function over the slice, spawning a new
+// goroutine for each element in the slice and then awaits all results before
+// returning them.
+func ForEachConc[A, B any](f func(A) B, as []A) []B {
```

Comment:
Perhaps we can lift this version? It uses an `errgroup` to make it easier to manage error result values: https://github.com/lightninglabs/taproot-assets/blob/main/fn/concurrency.go#L10-L34

Reply:
So I went to try and do this. There's trouble with going this route though. The code you link has the semantics where the supplied function definitely has an error result as a possibility and that the entire operation should be cancelled if any of them return an error. This has different semantics wherein the operation is either expensive and can benefit from multiprocessing, OR is IO bound and we want to do all of the work that we can and collect the results.

I believe there is room for both here. What I can do here is have this function also include a context object so it is still cancellable. I do not think that we can fully rewrite one of these in terms of the other, though.